src/util/png_image.hpp

```diff
@@ -0,0 +1,174 @@
+#pragma once
+
+#include "files.hpp"
+
+#include <png.h>
+
+#include <filesystem>
+#include <new>
+
+namespace image {
+
+enum write_result_t {
+    WR_OK = 0,
+    WR_FOPEN_FAILED,
+    WR_CREATE_WRITE_STRUCT_FAILED,
+    WR_CREATE_INFO_STRUCT_FAILED,
+    WR_SETJMP_PNG_JMPBUF_FAILED,
+    WR_ROW_MALLOC_FAILED,
+};
+
+template <class T>
```

Comment:
Make a concept `Image`

Reply:
Ya... There needs to be a number of concepts. Black-white, rgb-a etc. They are slightly different is that makes it hard.

Reply:
Start with just B&W for this PR